### ISSUE TITLE
Add kvm or lxd machines through juju_machine resource

### ISSUE THREAD

**Issue by arif-ali** [juju/terraform-provider-juju#347]
### Description

On an standard environment we normally can create kvm and/or lxd machines using the following method

```
juju add-machine kvm:3
juju add-machine lxd:3
```

There doesn't seem like a way to do this currently via the `juju_machine` resource

### Urgency

Casually reporting

### Terraform Juju Provider version

0.10.0

### Terraform version

1.6.4-dev

### Reproduce / Test

```shell
N/A
```

**Comment by amandahla** [comment]
@cderici Can I work on this one? I'm considering adding the parameter "placement" to the Machine resource.

**Comment by hmlanigan** [comment]
@amandahla, yes, your contributions are welcome. I can assign this to you. Please read the contributing section on the main page.

We do have a known problem to address around placement. Juju placement directives and actual machine numbers do not match. We're trying to understand how to handle this in the application resource as well. We need to be careful with placement in this case. 

Let's collaborate on the schema before you get deep into this.
